content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,213 @@
+metaTitle: 'Tracing (Preview)'
+metaDescription: 'Diagnose application performance with detailed traces of each query.'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+Prisma tracing logs
+Prisma tracing outputs
+
+Prisma tracing provides a detailed log of the activity that Prisma Client carries out, at a query level, including the time taken to execute each query. It helps you analyze your application's performance and identify bottlenecks. Tracing is fully compliant with [OpenTelemetry](https://opentelemetry.io/), so you can use it as part of your end-to-end application tracing path.
+
+<Admonition type="info">
+
+Tracing gives you a highly detailed, query-level insight into your Prisma project. If you want aggregated numerical reporting, such as query counts, connection counts, and total query execution times, see [Metrics](/concepts/components/prisma-client/metrics).
+
+</Admonition>
+
+</TopBlock>
+
+## About tracing
+
+When you enable tracing, Prisma Client writes out a trace for each request that it makes. **should we say operation here, rather than request**? Each trace includes one or more [spans](https://www.jaegertracing.io/docs/1.35/architecture/#span), each of which represents one stage in the request life-cycle (**check terminology here**), such as a database statement or command, or serialization and deserialization. Each span includes the round trip connection time from Prisma Client to the database and back.
+
+**Maybe a diagram here to show:**
+
+- Trace (Prisma operation)
+
+  - Span (such as a connection)
+  - Span (such as a database statement or command)
+  - Span (such as deserialization)
+
+  You can send tracing output to the console, or analyze it in any OpenTelemetry-compatible tracing system, such as [Jaeger](https://www.jaegertracing.io/).
+
+**TBA: maybe a screenshot in Jaeger with example**
+
+## Output
+
+Depending on the Prisma operation (**request?**), Prisma Client might output any of the following spans:
+
+- `prisma`: Represents the entire Prisma request, from Prisma Client to the database and back. It contains details such as the model and method called by the Prisma Client.
+- `prisma:query_builder`: Represents how long the [Prisma engines](/concepts/components/prisma-engines) take to generate the SQL queries (**would it be more accurate to say "statement or command" instead of queries here? This is not running rhe command, is it? - it's building the query**). This includes deserialization of the Prisma Client request, validation, and transformation into SQL.
+- `prisma:connection`: Represents how long it took for Prisma Client to get a database connection.
+- `prisma:db_query`: Represents a SQL query that was executed against the database, the SQL statement or command (in the tags), and how long the query took to run.
+- `prisma:itx_runner`: The start of an interactive transaction.
+- `prisma:transaction`: The high-level transaction. **need more detail here - what do we mean by high-level?**
```

Comment:
I think a picture could be nice for each of these (normal op vs. interactive transaction). If it's just in a list like this, it's hard to know which traces are applicable, how often they occur, etc.

If I just read a list like this, I envision an inverted pyramid where each one of these spans takes less time than the one above. That's the wrong mental image though.

Reply:
I've changed the list's intro text to make it clearer that any or all of these spans might appear, depending on the Prisma operation. 

I agree that one or more screenshots would help - I'll add some.

Reply:
I think there are 2 fundamental types: Queries and Transactions (with queries). Might be worth showing the normal Query first, and then a transaction that wraps multiple queries.